src/transformers/generation/candidate_generator.py

```diff
@@ -180,6 +186,10 @@ def __init__(
         # We need to roll back the cache in assisted generation, only DynamicCache is supported
         self.generation_config.cache_implementation = None
 
+        if self.assistant_model.generation_config.assistant_confidence_threshold:
```

Comment:
This block can be conditional to `is_sklearn_available`, correct? (if we don't have sklearn we don't adapt the threshold, and therefore we don't need to store this data)